parser/declaration.go

```diff
@@ -910,7 +910,10 @@ func parseHexadecimalLocation(p *parser) common.AddressLocation {
 	address, err := common.BytesToAddress(rawAddress)
 	if err != nil {
 		// Any returned error is a syntax error. e.g: Address too large error.
-		p.reportSyntaxError(err.Error())
```

Comment:
Do you know what was the problem with the old way?

Reply:
Looks like the linting got smarter: `reportSyntaxError()` has a "printf"-style signature `message string, params ...any`, and in turn calls `syntaxError()`, which has a similar signature, and that in turn calls `NewSyntaxError`, which again has a similar signature, and eventually calls `fmt.Sprintf(message, params...)`.

So `err.Error()` is used here as a format string, but really should be used as-is as the message.

Reply:
hah, a bit too smart! 😆